src/Chapter4.hs

```diff
@@ -629,7 +666,19 @@ Can you implement a monad version of AND, polymorphic over any monad?
 🕯 HINT: Use "(>>=)", "pure" and anonymous function
 -}
 andM :: (Monad m) => m Bool -> m Bool -> m Bool
-andM = error "andM: Not implemented!"
+andM m1 m2 =
+    m1 >>= (\x -> m2 >>= (\y -> pure $ x && y))
+
+andM2 :: (Monad m) => m Bool -> m Bool -> m Bool
+andM2 m1 m2 =
+    m1 >>= (\x -> m2 >>= (\y -> return $ x && y))
+
+andM3 :: (Monad m) => m Bool -> m Bool -> m Bool
+andM3 m1 m2 = do
+    x <- m1
+    y <- m2
+    return $ x && y
```

Comment:
Great implementations! All these functions are semantically equivalent ↔️ 

Though, your current implementation doesn't pass all tests. The ordinary `&&` if lazy on the second argument. Meaning that if it meets `False` as the first argument, there is no need to calculate another arg. In `andM` we would like to keep the same behaviour, so the

```
Just False `andM` _
```

should be `Just False` :slightly_smiling_face: